typescript/sdk/src/crud/EvmIsmModule.ts

```diff
@@ -1,52 +1,89 @@
+import { EvmIsmCreator } from '../ism/EvmIsmCreator.js';
 import { EvmIsmReader } from '../ism/read.js';
 import { IsmConfig } from '../ism/types.js';
 import { MultiProvider } from '../providers/MultiProvider.js';
 import { EthersV5Transaction } from '../providers/ProviderType.js';
+import { ChainNameOrId } from '../types.js';
 
 import { CrudModule, CrudModuleArgs } from './AbstractCrudModule.js';
 
 // WIP example implementation of EvmIsmModule
 export class EvmIsmModule extends CrudModule<
   ProtocolType.Ethereum,
   IsmConfig,
-  HyperlaneAddresses<ProxyFactoryFactories>
+  HyperlaneContracts<ProxyFactoryFactories> & {
+    deployedIsm: Address;
+  }
 > {
   protected logger = rootLogger.child({ module: 'EvmIsmModule' });
   protected reader: EvmIsmReader;
+  protected creator: EvmIsmCreator;
 
   protected constructor(
     protected readonly multiProvider: MultiProvider,
-    args: Omit<
-      CrudModuleArgs<
-        ProtocolType.Ethereum,
-        IsmConfig,
-        HyperlaneAddresses<ProxyFactoryFactories>
-      >,
-      'provider'
+    protected readonly deployer: HyperlaneDeployer<any, any>,
+    args: CrudModuleArgs<
+      IsmConfig,
+      HyperlaneContracts<ProxyFactoryFactories> & {
+        deployedIsm: Address;
+      }
     >,
   ) {
-    super({
-      ...args,
-      provider: multiProvider.getProvider(args.chain),
-    });
-
+    super(args);
     this.reader = new EvmIsmReader(multiProvider, args.chain);
+    this.creator = new EvmIsmCreator(deployer, multiProvider, args.addresses);
   }
 
-  public async read(address: Address): Promise<IsmConfig> {
-    return await this.reader.deriveIsmConfig(address);
+  public async read(): Promise<IsmConfig> {
+    return await this.reader.deriveIsmConfig(this.args.addresses.deployedIsm);
   }
 
-  public async update(_config: IsmConfig): Promise<EthersV5Transaction[]> {
+  public async update(config: IsmConfig): Promise<EthersV5Transaction[]> {
     throw new Error('Method not implemented.');
+
+    const destination = this.multiProvider.getChainName(this.args.chain);
+    await this.creator.update({
+      destination,
+      config,
+      // origin?: string | undefined;
+      // mailbox?: string | undefined;
+      existingIsmAddress: this.args.addresses.deployedIsm,
+    });
+    return [];
   }
 
   // manually write static create function
-  public static create(_config: IsmConfig): Promise<EvmIsmModule> {
-    throw new Error('not implemented');
+  public static async create({
+    chain,
+    config,
+    deployer,
+    factories,
+    multiProvider,
+  }: {
+    chain: ChainNameOrId;
+    config: IsmConfig;
+    deployer: HyperlaneDeployer<any, any>;
+    factories: HyperlaneContracts<ProxyFactoryFactories>;
+    multiProvider: MultiProvider;
+  }): Promise<EvmIsmModule> {
+    const destination = multiProvider.getChainName(chain);
+    const ismCreator = new EvmIsmCreator(deployer, multiProvider, factories);
```

Comment:
The architecture here is confusing me a bit. So EvmIsmModule has a static create method which actually just uses an EvmIsmCreator class, though not statically. 

Can you clarify why an EvmIsmCreator is needed?

Reply:
EvmIsmCreator is a fork of the existing HyperlaneIsmFactory, made to be standalone without any HyperlaneApp reliance, it was a quick way to get the functionality of updating and deploying without reinventing the wheel

I didn't get the chance to work on integrating the methods themselves further with the EvmIsmModule, as I moved to prepare a barebones version of this class that could unblock @ltyu. What you see at the moment is an intermediary stage that should resolve itself as part of the `EvmIsmModule.update()` implementation